diff --git a/igvm-tools/src/builder.rs b/igvm-tools/src/builder.rs
index 61e9fa7..55076ba 100644
--- a/igvm-tools/src/builder.rs
+++ b/igvm-tools/src/builder.rs
@@ -352,7 +352,16 @@ impl Builder {
 
     pub fn finalize(&self) -> Result<IgvmFile, igvm::Error> {
         let mut directives = self.directives.clone();
-        directives.sort_by(Self::sort_pages);
+        directives.sort_by(|a, b| match (a, b) {
+            (
+                IgvmDirectiveHeader::SnpVpContext { vp_index: a, .. },
+                IgvmDirectiveHeader::SnpVpContext { vp_index: b, .. },
+            ) => a.cmp(b),
+            (IgvmDirectiveHeader::SnpVpContext { .. }, _) => Ordering::Greater,
+            (_, IgvmDirectiveHeader::SnpVpContext { .. }) => Ordering::Less,
+            _ => Self::sort_pages(a, b),
+        });
+
         IgvmFile::new(
             self.revision,
             self.platforms.clone(),
